static/js/lib/validation/profile.js

```diff
@@ -299,5 +299,10 @@ export const validateFinancialAid = (edit: FinancialAidState): FinancialAidValid
   if (!edit.checkBox) {
     errors['checkBox'] = 'You must agree to these terms';
   }
+
+  if (edit.income && !/^\d+$/.test(edit.income)) {
```

Comment:
I don't think you don't need the quantifiers or the anchors (`^`, `+`, `$`) either because all we care about is whether there is a non-digit character anywhere in the string